src/app/query/bitcoin/transaction/use-bitcoin-broadcast-transaction.ts

```diff
@@ -18,8 +19,20 @@ export function useBitcoinBroadcastTransaction() {
   const analytics = useAnalytics();
 
   const broadcastTx = useCallback(
-    async ({ tx, onSuccess, onError, onFinally, delayTime = 700 }: BroadcastCallbackArgs) => {
+    async ({
+      tx,
+      onSuccess,
+      onError,
+      onFinally,
+      delayTime = 700,
+      hasPassedCheckForInscribedUtxos,
+    }: BroadcastCallbackArgs) => {
       try {
+        // add explicit check in broadcastTx to ensure that utxos are checked before broadcasting
+        if (!hasPassedCheckForInscribedUtxos) {
+          throw new Error('Utxos should be checked for inscriptions before broadcasting');
+        }
```

Comment:
This seems like an odd API to me, why not either:

1. Conditionally invoke this function based on outcome of `hasPassedCheckForInscribedUtxos`, rather than passing it in
2. Perform the check from within the broadcast function?

Reply:
yeah. I understand that this looks odd, but I added it so intentionally in order not to forget in other (future) flows to add utxos check. this will prevent from missing it.

also I haven't added it inside the broadcast fn, bc I think this should be responsible for broadcasting, and not adding some other logic (e.g. we will need to pass filtered utxos list arg here (filtered from inscription utxo being sent in inscription send flow)).